extension/src/paymentHandler/web-component-service.js

```diff
@@ -0,0 +1,50 @@
+const fetch = require('node-fetch')
+const { serializeError } = require('serialize-error')
+
+const configLoader = require('../config/config')
+const config = configLoader.load()
+
+function getOriginKeys (getOriginKeysRequest) {
+  return callAdyen('originKeys', getOriginKeysRequest)
+}
+
+function getPaymentMethods (getPaymentMethodsRequest) {
+  return callAdyen('paymentMethods', getPaymentMethodsRequest)
+}
+
+async function callAdyen(endpoint, requestString) {
+  let request = buildRequest(requestString)
+  let response
+  try {
+    response = await fetchAsync(endpoint, request)
+  } catch (err) {
+    response = serializeError(err)
+  }
+  return { request, response }
+}
+
+function buildRequest (requestString) {
+  const body = JSON.stringify({
+    merchantAccount: config.adyen.merchantAccount,
+    ...JSON.parse(requestString)
+  })
+
+  return {
+    method: 'POST',
+    body: body,
+    headers: {
+      'Content-Type': 'application/json',
+      'X-Api-Key': config.adyen.apiKey
```

Comment:
Is it removed before we persist it as interface interaction?

Reply:
yes, it is. but that part is not clear from our implementations (was not clear before too), maybe I should improve the design of creating actions both for interface interaction and custom fields.

Reply:
Where is it removed? I would expect that there is a single place where both request and response is stripped from sensitive data.

Reply:
I am not removing it because the data is stored as a header, persisting only the body is enough.

Reply:
>I would expect that there is a single place where both request and response is stripped from sensitive data.

I moved those parts from utils to web-components-service file, so it's now in one place, also all adyen calls are in one place.